kong/tracing/instrumentation.lua

```diff
@@ -201,12 +199,12 @@ function _M.request(ctx)
     http_flavor = string.format("%.1f", http_flavor)
   end
 
-  local active_span = tracer.start_span(span_name, {
+  local active_span = tracer.start_span("kong.request", {
```

Comment:
'kong.request' is a bit generic, what activities happen under this span?

Reply:
@hbagdi  This span hold the information (method, path, scheme, host, ip, ...) of any http request proxied by Kong. This is similar to `nginx.request`, see the default behaviour of the [Datadog Nginx tracing module](https://github.com/DataDog/nginx-datadog/#default-behavior).

This PR is still open exactly to allow further discussion to finalize these names, please feel free to suggest a better alternative.

Reply:
If this is the "root" span of Kong, meaning, this is the parent span of all spans that Kong generates, then simply calling it 'kong' would be better.
The reason I say that is most, if not all, users use tracing with multiple services and kong is one such service in such a case. When looking at a trace, the topmost span generally for any service (kong is a service in this sentence), is generally the service name since that span tells you how long did everything take in that span.
Now, the upstream spans will also be part of the 'kong' span in such a case since it is most likely that the traces generated in upstream inherit the span started by kong.

ThoughtS?

Reply:
That sounds right. Since every trace always represents a request, the word `request` is probably redundant for the root span. I think it's also safe to say that this will always be the root span (it will never have a parent), so the name `kong` probably makes the most sense. If I recall correctly @kikito was also suggesting the same? @StarlightIbuki do you also agree or is there anything we are missing?

Reply:
@samugi I once use "proxy" as the name of the root span, and then "request". The "kong." is rather a prefix introduced because of the customer's requirement.

I agree this seems redundant. But if we want to change the naming schema, let's also rename other spans to make it consistent (so at least we can convince the customer).

Reply:
I think what Harry meant (that I was agreeing with) is that this being the root span, it can only contain the "prefix", so `kong` alone could be a good name for this span, while leaving the rest unchanged. Am I right @hbagdi?

Reply:
This explanation makes sense to me.